pkg/execution/execution.go

```diff
@@ -166,34 +167,53 @@ func newScenarioInfo(getCtx func() context.Context) (*execInfo, error) {
 	return newExecInfo(rt, si), nil
 }
 
-// newTestInfo returns a goja.DynamicObject implementation to retrieve
-// information about the overall test run (local instance).
-func newTestInfo(getCtx func() context.Context) (*execInfo, error) {
-	ctx := getCtx()
+func getES(ctx context.Context, rt *goja.Runtime) *lib.ExecutionState {
 	es := lib.GetExecutionState(ctx)
 	if es == nil {
-		return nil, errors.New("getting test information in the init context is not supported")
+		common.Throw(rt, errors.New("getting test information in the init context is not supported"))
 	}
+	return es
+}
 
+// newTestInfo returns a goja.DynamicObject implementation to retrieve
+// information about the overall test run (local instance), and to allow
+// stopping it with abort().
+func newTestInfo(getCtx func() context.Context) (*execInfo, error) {
+	ctx := getCtx()
 	rt := common.GetRuntime(ctx)
 	if rt == nil {
 		return nil, errors.New("goja runtime is nil in context")
 	}
 
 	ti := map[string]func() interface{}{
+		// stop the test run
+		"abort": func() interface{} {
+			return func(msg goja.Value) {
```

Comment:
There's a slight issue with this approach.

This script:

```javascript
import exec from 'k6/x/execution';

exec.test.abort;

export default function () {
  console.log(exec.test.duration);
}
```

Outputs:

```
ERRO[0000] getting test information in the init context is not supported
        at file:///tmp/test.js:6:15(7)  executor=per-vu-iterations scenario=default source=stacktrace
```

This happens on the `test.duration` access, so it seems that accessing `test.abort` in `init` caches the `nil` `*ExecutionState`. Actually calling `test.abort()` or commenting out that line works as expected.

Reply:
I do think that you should just move to getting the `ctx` and `rt` in each of the callbacks so it's always just get the current one. Hopefully with the new Modules this will be even cheaper and already the whole goja reflection parts are probably *way* more expensive

Reply:
So with the rebase on #2 this is no longer an issue :tada:

I'm not sure if it's because of the `ModuleV2` change or the removal of `Proxy`/`DynamicObject`, but I didn't want to do any additional changes, as it doesn't seem necessary to get `ctx` and `rt` in each function. We do need a fresh `ExecutionState` though, but that was required before this update.